enhancements/storage/csi-migration.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Migration of in-tree volume plugins to CSI drivers
+authors:
+  - "@fbertina"
+reviewers:
+  - "@openshift/storage ”
+approvers:
+  - "@openshift/openshift-architects"
+  - "@darkmuggle"
+creation-date: 2020-07-01
+last-updated: 2021-01-28
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Migration of in-tree volume plugins to CSI drivers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+We want to allow cluster administrators to seamlessly migrate in-tree volumes to their counterparts CSI drivers. It is important to achieve this goal before CSI Migration feature becomes GA in upstream.
+
+## Motivation
+
+CSI migration is going to be enabled by default as a beta feature in Kubernetes 1.21 (OCP 4.8). In Kubenertes 1.22 (OCP 4.9) the feature may become GA.
+
+In OCP we can optionally disable beta features, however, that will no longer be an option once CSI migration becomes GA.
+
+In order to avoid surprises once the migration is enabled by default in OCP, we want to allow cluster administrators to optionally enable the feature earlier, preferably in OCP 4.8.
+
+## Goals
+
+For Tech Preview, we want to introduce a mechanism to allow switching CSI migration feature flags on and off accross OCP components.
+
+Due to upstream requirements, it is important that this mechanism allows for enabling the feature flags on control-plane components **before** the kubelet, and vice-versa.
+
+Once CSI migration is enabled by default in upstream, it will not be possible to disable it again. Therefore, such mechanism shall be disabled in OCP once CSI migration becomes GA in upstream.
+
+## Non-Goals
+
+* Control the ordering in which OCP components will be upgraded or downgraded. We will leave this responsability to the user.
```

Comment:
User have no control over ordering of OCP components, I'd drop that last sentence from here.